blockprotocol/blockprotocol
# Add BP reporting og block listing

## 🌟 What is the purpose of this PR?



This PR adds analytics reporting for listing blocks. We want to see who/when/from where blocks were fetched.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203623179643284/1203983951760415/f) _(internal)_

## 🚫 Blocked by





## 🔍 What does this change?





## 📜 Does this require a change to the docs?


- I've added documentation about the new environment variables

## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?


- manual testing, I've ensured the shape matches what is expected in the data collector service we run.

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Set the follow values in your local .env
```
DATA_URL=https://requestinspector.com/inspect/01grv082jp6z0yk13tjxrx2c20
DATA_WRITE_KEY=something-not-relevant-for-test
```
1.  Call the `/api/blocks` endpoint with an API key
1.  observe that the data is reported to the endpoint at https://requestinspector.com/p/01grv082jp6z0yk13tjxrx2c20

## 📹 Demo



![image](https://user-images.githubusercontent.com/6988668/219363572-8d6823b6-cd89-475c-ba19-0124b4af7413.png)
